test/cpp/ext/otel/grpc_trace_bin_text_map_propagator_test.cc

```diff
@@ -0,0 +1,122 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#include <grpcpp/ext/otel_plugin.h>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "opentelemetry/trace/context.h"
+#include "src/core/lib/transport/metadata_batch.h"
+#include "src/cpp/ext/otel/otel_plugin.h"
+#include "test/core/promise/test_context.h"
+#include "test/core/test_util/test_config.h"
+
+namespace grpc {
+namespace testing {
+namespace {
+
+class TestTextMapCarrier
+    : public opentelemetry::context::propagation::TextMapCarrier {
+ public:
+  opentelemetry::nostd::string_view Get(
+      opentelemetry::nostd::string_view key) const noexcept override {
+    if (key == "grpc-trace-bin") {
+      return value_;
+    } else {
+      return "";
+    }
+  }
+
+  void Set(opentelemetry::nostd::string_view key,
+           opentelemetry::nostd::string_view value) noexcept override {
+    if (key == "grpc-trace-bin") {
+      value_ = std::string(value);
+    }
+  }
+
+ private:
+  std::string value_;
+};
+
+TEST(GrpcTraceBinTextMapPropagatorTest, Inject) {
+  auto propagator = experimental::MakeGrpcTraceBinTextMapPropagator();
+  TestTextMapCarrier carrier;
+  opentelemetry::context::Context context;
+  char trace_id[] = "0123456789ABCDEF";
+  char span_id[] = "01234567";
+  context = opentelemetry::trace::SetSpan(
+      context,
+      std::shared_ptr<opentelemetry::trace::Span>(
+          new (std::nothrow) opentelemetry::trace::DefaultSpan(
+              opentelemetry::trace::SpanContext(
+                  opentelemetry::trace::TraceId(
+                      opentelemetry::nostd::span<const uint8_t, 16>(
+                          reinterpret_cast<const uint8_t*>(trace_id), 16)),
+                  opentelemetry::trace::SpanId(
+                      opentelemetry::nostd::span<const uint8_t, 8>(
+                          reinterpret_cast<const uint8_t*>(span_id), 8)),
+                  opentelemetry::trace::TraceFlags(1), /*is_remote=*/true))));
+  propagator->Inject(carrier, context);
+  std::string unescaped_val;
+  absl::Base64Unescape(
+      internal::NoStdStringViewToAbslStringView(carrier.Get("grpc-trace-bin")),
+      &unescaped_val);
+  EXPECT_EQ(unescaped_val[0], 0);
+  EXPECT_EQ(unescaped_val[0], 0);
+  EXPECT_EQ(absl::string_view(unescaped_val).substr(2, 16), trace_id);
+  EXPECT_EQ(unescaped_val[18], 1);
+  EXPECT_EQ(absl::string_view(unescaped_val).substr(19, 8), span_id);
+  EXPECT_EQ(unescaped_val[27], 2);
+  EXPECT_EQ(unescaped_val[28], 1);
+}
+
+TEST(GrpcTraceBinTextMapPropagatorTest, Extract) {
+  TestTextMapCarrier carrier;
+  std::array<char, 29> trace_bin_val;
```

Comment:
I think this can be written in a way that doesn't require hard-coding the size:

```
constexpr char kTraceBinValue[] =
    "\x00\x00"
    "0123456789ABCDEF"
    "\x01"
    "01234567\x02\x01";
absl::string_view trace_bin_value(kTraceBinValue, sizeof(kTraceBinValue));
```